wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/ServiceTerminator.java

```diff
@@ -44,16 +55,109 @@
   void terminateAll() throws InterruptedException {
```

Comment:
maybe declare in docs that the termination is carried out in the order that is determined by the presence of dependencies between service stoppers, wdyt?

Reply:
\+